### DISCUSSION TITLE
Using environment variables as dynamic keys

### DISCUSSION THREAD

**Discussion by mudideng** [mikefarah/yq#764] [Q&A] [answered]
Hi! In yq v3, I used to be able to do set values to keys dynamically, something like
```
yq write --inplace \
              "${SOME_FILE}" \
              "animals.${FAMILY}" \
              "${SPECIES}"
```

Is this still possible with yq v4? I couldn't figure out a way to do it. The `env()` operator only seems to work for setting the value when I tested it.

**Comment by mikefarah** [comment, marked as answer]
Yep, you can do it using interpolation in your shell like you've done, or via the env operator documented here: https://mikefarah.gitbook.io/yq/operators/env-variable-operators#dynamic-key-lookup-with-environment-variable

Env operator is useful just because it sidesteps escaping and quoting issues in shells

**Reply by mudideng** [reply]
Thanks for the answer! Kinda embarrassing that I searched for dynamic keys and still didn't find it when it's right there in the documentation...

And indeed, maybe I should use the env operator, since the issue I had was because of quoting issues. Thanks again!
